### ISSUE TITLE
Crash in Xcode version 9.0 (9A235) showing PKHUDProgressView

### ISSUE THREAD

**Issue by alexwillingham** [pkluz/PKHUD#196]
Xcode version 9.0 (9A235)
Swift 3.2 
Running on iPhone X Simulator

Line of code executed:

`HUD.show(.labeledProgress(title: "Loading contacts...", subtitle: nil)) })`

Log of crash:
```
2017-09-15 16:57:57.852845-0400 redactedAppName[31636:558237] *** Assertion failure in -[PKHUD.FrameView _addSubview:positioned:relativeTo:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKit_Sim/UIKit-3694.4.18/UIVisualEffectView.m:1464
2017-09-15 16:57:57.880328-0400 redactedAppName[31636:558237] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: '<PKHUD.PKHUDProgressView: 0x7fa3a4597630; frame = (0 0; 156 156); clipsToBounds = YES; alpha = 0.85; layer = <CALayer: 0x60000042df60>> has been added as a subview to <PKHUD.FrameView: 0x7fa3a4578580; baseClass = UIVisualEffectView; frame = (0 0; 156 156); clipsToBounds = YES; autoresize = LM+RM+TM+BM; layer = <CALayer: 0x60000042a820>>. Do not add subviews directly to the visual effect view itself, instead add them to the -contentView.'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000011083d1cb __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010fa2cf41 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000110842362 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x000000010bc01089 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 193
	4   UIKit                               0x000000010ded243f -[UIVisualEffectView _addSubview:positioned:relativeTo:] + 379
	5   PKHUD                               0x000000010d1c04a5 _T05PKHUDAAC11contentViewSo6UIViewCfsTf4gn_n + 373
	6   PKHUD                               0x000000010d1bc06f _T05PKHUD3HUDC4showyAA14HUDContentTypeOFZ + 303
	7   redactedAppName                           0x00000001096e6fd8 _T09redactedAppName22ContactsViewControllerC04loadC07RxSwift10ObservableCyAA0C5EventOGyFySbcfU_ + 152
	8   redactedAppName                           0x00000001096e6ff9 _T0Sbs5Error_pIxyzo_SbsAA_pIxizo_TR + 25
	9   redactedAppName                           0x00000001096f1df3 _T0Sbs5Error_pIxyzo_SbsAA_pIxizo_TRTA + 83
	10  RxSwift                             0x000000010d460482 _T07RxSwift14ObservableTypePAAE2doAA0C0Cy1EQzGyAHKcSg6onNext_ys5Error_pKcSg0fH0yyKcSg0F9CompletedyycSg0F9SubscribeAQ0F7DisposetFyAA5EventOyAHGKcfU_ + 370
	11  RxSwift                             0x000000010d460662 _T07RxSwift14ObservableTypePAAE2doAA0C0Cy1EQzGyAHKcSg6onNext_ys5Error_pKcSg0fH0yyKcSg0F9CompletedyycSg0F9SubscribeAQ0F7DisposetFyAA5EventOyAHGKcfU_TA + 162
	12  RxSwift                             0x000000010d44e735 _T07RxSwift6DoSinkC2onyAA5EventOy1EQzGF + 277
	13  RxSwift                             0x000000010d44eac4 _T07RxSwift6DoSinkCyxGAA12ObserverTypeA2aERzlAaEP2onyAA5EventOy1EQzGFTW + 36
	14  RxSwift                             0x000000010d463c91 _T07RxSwift32ObserveOnSerialDispatchQueueSinkCACyxGAA0efG9SchedulerC9scheduler_x8observerAA10Cancelable_p6canceltcfcAA10Disposable_pAD_AA5EventOy1EQzGtcfU_ + 241
	15  RxSwift                             0x000000010d464659 _T07RxSwift32ObserveOnSerialDispatchQueueSinkCACyxGAA0efG9SchedulerC9scheduler_x8observerAA10Cancelable_p6canceltcfcAA10Disposable_pAD_AA5EventOy1EQzGtcfU_TA + 9
	16  RxSwift                             0x000000010d46470c _T07RxSwift32ObserveOnSerialDispatchQueueSinkCyxGAA5EventOy1EQzGAA10Disposable_pIxxir_AD_AItAaJ_pIxir_AA12ObserverTypeRzlTRTA + 140
	17  RxSwift                             0x000000010d457bea _T07RxSwift13MainSchedulerC16scheduleInternalAA10Disposable_px_AaE_pxc6actiontlF + 250
	18  RxSwift                             0x000000010d48322b _T07RxSwift28SerialDispatchQueueSchedulerC8scheduleAA10Disposable_px_AaE_pxc6actiontlF + 75
	19  RxSwift                             0x000000010d463fb6 _T07RxSwift32ObserveOnSerialDispatchQueueSinkC6onCoreyAA5EventOy1EQzGF + 566
	20  RxSwift                             0x000000010d4648e3 _T07RxSwift12ObserverBaseC2onyAA5EventOyxGF + 403
	21  RxSwift                             0x000000010d464b94 _T07RxSwift12ObserverBaseCyxGAA0C4TypeAAlAaEP2onyAA5EventOy1EQzGFTW + 36
	22  RxSwift                             0x000000010d456891 _T07RxSwift4JustC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 161
	23  RxSwift                             0x000000010d464456 _T07RxSwift28ObserveOnSerialDispatchQueueC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6cancelt1EQyd__RszAA12ObserverTypeRd__lF + 198
	24  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	25  RxSwift                             0x000000010d44ecb5 _T07RxSwift2DoC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6cancelt1EQyd__RszAA12ObserverTypeRd__lF + 261
	26  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	27  RxSwift                             0x000000010d45b070 _T07RxSwift9MergeSinkC3runAA10Disposable_pAA10ObservableCyxGF + 432
	28  RxSwift                             0x000000010d45c757 _T07RxSwift7FlatMapC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6canceltAA12ObserverTypeRd__1EQyd__AKRt_lFTm + 263
	29  RxSwift                             0x000000010d45c148 _T07RxSwift7FlatMapC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6canceltAA12ObserverTypeRd__1EQyd__AKRt_lF + 24
	30  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	31  RxSwift                             0x000000010d45b070 _T07RxSwift9MergeSinkC3runAA10Disposable_pAA10ObservableCyxGF + 432
	32  RxSwift                             0x000000010d45c757 _T07RxSwift7FlatMapC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6canceltAA12ObserverTypeRd__1EQyd__AKRt_lFTm + 263
	33  RxSwift                             0x000000010d45c148 _T07RxSwift7FlatMapC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6canceltAA12ObserverTypeRd__1EQyd__AKRt_lF + 24
	34  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	35  RxSwift                             0x000000010d459132 _T07RxSwift3MapC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6cancelt1EQyd__Rs_AA12ObserverTypeRd__lF + 242
	36  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	37  RxSwift                             0x000000010d425188 _T07RxSwift9CatchSinkC3runAA10Disposable_pyF + 952
	38  RxSwift                             0x000000010d425aa4 _T07RxSwift5CatchC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6cancelt1EQyd__RszAA12ObserverTypeRd__lF + 100
	39  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	40  RxSwift                             0x000000010d464456 _T07RxSwift28ObserveOnSerialDispatchQueueC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6cancelt1EQyd__RszAA12ObserverTypeRd__lF + 198
	41  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	42  RxSwift                             0x000000010d44ecb5 _T07RxSwift2DoC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6cancelt1EQyd__RszAA12ObserverTypeRd__lF + 261
	43  RxSwift                             0x000000010d467f85 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 389
	44  RxSwift                             0x000000010d464456 _T07RxSwift28ObserveOnSerialDispatchQueueC3runAA10Disposable_p4sink_AaE_p12subscriptiontqd___AA10Cancelable_p6cancelt1EQyd__RszAA12ObserverTypeRd__lF + 198
	45  RxSwift                             0x000000010d4682cc _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lFAaE_pycfU_ + 364
	46  RxSwift                             0x000000010d468c93 _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lFAaE_pycfU_TA + 83
	47  RxSwift                             0x000000010d4407ef _T07RxSwift22CurrentThreadSchedulerC8scheduleAA10Disposable_px_AaE_pxc6actiontlFyt_Tg504_T07a7Swift10G20_pIxr_ytAaB_pIxir_TRAaE_pIxr_Tf1nncn_n + 527
	48  RxSwift                             0x000000010d4680dd _T07RxSwift8ProducerC9subscribeAA10Disposable_pqd__1EQyd__RszAA12ObserverTypeRd__lF + 733
	49  RxSwift                             0x000000010d461c82 _T07RxSwift14ObservableTypePAAE9subscribeAA10Disposable_py1EQzcSg6onNext_ys5Error_pcSg0gI0yycSg0G9CompletedAM0G8DisposedtF + 690
	50  redactedAppName                           0x00000001096e1e49 _T09redactedAppName22ContactsViewControllerC04loadC15AndHandleResult33_16DC83E487ECD90299D7E27593AF19EFLLyyF + 489
	51  redactedAppName                           0x00000001096dd17a _T09redactedAppName22ContactsViewControllerC6handleyAA0C5EventO5event_tF + 15098
	52  redactedAppName                           0x00000001096d95e6 _T09redactedAppName22ContactsViewControllerCAcA0C5PropsV5props_tcfcySayypGcfU_ + 70
	53  redactedAppName                           0x00000001096d967a _T09redactedAppName22ContactsViewControllerCAcA0C5PropsV5props_tcfcySayypGcfU_TA + 58
	54  redactedAppName                           0x00000001096dd1ff _T0SayypGIxx_AAIxi_TR + 15
	55  redactedAppName                           0x00000001096dd292 _T0SayypGIxx_AAIxi_TRTA + 66
	56  RxSwift                             0x000000010d462078 _T07RxSwift14ObservableTypePAAE9subscribeAA10Disposable_py1EQzcSg6onNext_ys5Error_pcSg0gI0yycSg0G9CompletedAM0G8DisposedtFyAA5EventOyAGGcfU_Tf4ngggg_n + 312
	57  RxSwift                             0x000000010d461db9 _T07RxSwift14ObservableTypePAAE9subscribeAA10Disposable_py1EQzcSg6onNext_ys5Error_pcSg0gI0yycSg0G9CompletedAM0G8DisposedtFyAA5EventOyAGGcfU_ + 57
	58  RxSwift                             0x000000010d461f2c _T07RxSwift14ObservableTypePAAE9subscribeAA10Disposable_py1EQzcSg6onNext_ys5Error_pcSg0gI0yycSg0G9CompletedAM0G8DisposedtFyAA5EventOyAGGcfU_TA + 172
	59  RxSwift                             0x000000010d41a277 _T07RxSwift17AnonymousObserverC6onCoreyAA5EventOyxGF + 39
	60  RxSwift                             0x000000010d4648e3 _T07RxSwift12ObserverBaseC2onyAA5EventOyxGF + 403
	61  RxSwift                             0x000000010d464b94 _T07RxSwift12ObserverBaseCyxGAA0C4TypeAAlAaEP2onyAA5EventOy1EQzGFTW + 36
	62  RxSwift                             0x000000010d488168 _T07RxSwift4SinkC9forwardOnyAA5EventOy1EQzGF + 120
	63  RxSwift                             0x000000010d492ea3 _T07RxSwift13TakeCountSinkC2onyAA5EventOy1EQzGF + 531
	64  RxSwift                             0x000000010d4930f4 _T07RxSwift13TakeCountSinkCyxGAA12ObserverTypeA2aERzlAaEP2onyAA5EventOy1EQzGFTW + 36
	65  RxSwift                             0x000000010d464ee2 _T0TA + 82
	66  RxSwift                             0x000000010d41a455 _T07RxSwift11AnyObserverV2onyAA5EventOyxGF + 37
	67  RxSwift                             0x000000010d41a4ab _T07RxSwift11AnyObserverVyxGAA0D4TypeAAlAaEP2onyAA5EventOy1EQzGFTW + 43
	68  RxSwift                             0x000000010d41ba3c _T07RxSwift3BagVA2A12ObserverTypeRzlE2onyAA5EventOy1EQzGF + 636
	69  RxSwift                             0x000000010d4690eb _T07RxSwift14PublishSubjectC2onyAA5EventOyxGF + 299
	70  RxCocoa                             0x000000010d2ba588 _T07RxCocoa16MessageSentProxy33_F7515DBB13B60709A3CB25DD19EDD11DLLC13methodInvokedySayypG13withArguments_tFTo + 120
	71  RxCocoa                             0x000000010d29b7c8 RX_forward_invocation + 236
	72  RxCocoa                             0x000000010d29b67b __71-[RXObjCRuntime(InfrastructureMethods) swizzleForwardInvocation:error:]_block_invoke_2 + 44
	73  CoreFoundation                      0x00000001107bfed8 ___forwarding___ + 760
	74  CoreFoundation                      0x00000001107bfb58 _CF_forwarding_prep_0 + 120
	75  UIKit                               0x000000010da889b0 -[UIViewController _setViewAppearState:isAnimating:] + 444
	76  UIKit                               0x000000010da89245 -[UIViewController __viewWillAppear:] + 147
	77  UIKit                               0x000000010dab522c -[UINavigationController _startCustomTransition:] + 1195
	78  UIKit                               0x000000010dacb894 -[UINavigationController _startDeferredTransitionIfNeeded:] + 686
	79  UIKit                               0x000000010daccb90 -[UINavigationController __viewWillLayoutSubviews] + 115
	80  UIKit                               0x000000010dd232ae -[UILayoutContainerView layoutSubviews] + 231
	81  UIKit                               0x000000010d9b3551 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1331
	82  QuartzCore                          0x00000001117254ba -[CALayer layoutSublayers] + 153
	83  QuartzCore                          0x00000001117295a9 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 401
	84  QuartzCore                          0x00000001116b21cd _ZN2CA7Context18commit_transactionEPNS_11TransactionE + 365
	85  QuartzCore                          0x00000001116ddae4 _ZN2CA11Transaction6commitEv + 500
	86  QuartzCore                          0x00000001116de830 _ZN2CA11Transaction17observer_callbackEP19__CFRunLoopObservermPv + 76
	87  CoreFoundation                      0x00000001107dfdb7 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	88  CoreFoundation                      0x00000001107dfd0e __CFRunLoopDoObservers + 430
	89  CoreFoundation                      0x00000001107c4324 __CFRunLoopRun + 1572
	90  CoreFoundation                      0x00000001107c3a89 CFRunLoopRunSpecific + 409
	91  GraphicsServices                    0x00000001140339c6 GSEventRunModal + 62
	92  UIKit                               0x000000010d8e4d30 UIApplicationMain + 159
	93  redactedAppName                           0x000000010976e997 main + 55
	94  libdyld.dylib                       0x00000001126b5d81 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by alexwillingham** [comment]
Hopefully the crash recommendation is sufficient?:  "Do not add subviews directly to the visual effect view itself, instead add them to the -contentView.'"

**Comment by PGLongo** [comment]
You should use the swift4 branch with Xcode 9

**Comment by alexwillingham** [comment]
Thanks for the reply @PGLongo.  I'd like to keep my project in Swift 3.2 for a while by setting `swift language version` to 3.2 in build settings, while using Xcode 9, so that I can still ship builds to the App Store.  And I'd upgrade my code to swift 4 at a more convenient time.  No plans to support Swift 3.2 in Xcode 9 then?

**Comment by PGLongo** [comment]
Unfortunately it seems to be an issue related to iOS 11 and not Swift.

If you want any to contribute with a PR any help is appreciated, I don't have enough time

**Comment by alexwillingham** [comment]
So this is our bad. We were building from an old branch of PKHUD. But you've patched this up all the way back in June! https://github.com/pkluz/PKHUD/pull/169 

Thanks for the help!

**Comment by PGLongo** [comment]
You are welcome

**Comment by PraveenPali** [comment]
Assertion failure in -[UIVisualEffectView _addSubview:positioned:relativeTo:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKit_Sim/UIKit-3694.4.18/UIVisualEffectView.m:1464

**Comment by PGLongo** [comment]
@PraveenPali this issue has been fixed with #199. Update your pod to the latest version
